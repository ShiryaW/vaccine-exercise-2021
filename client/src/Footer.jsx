import React, { Component } from "react";
import PropTypes from "prop-types";

export class Footer extends Component {
  constructor(props) {
    super(props);
  }

  render = () => {
    return (
      <div className="footer">
        <p>Total items: {this.props.totalItems}</p>
      </div>
    );
  };
}

Footer.propTypes = {
  totalItems: PropTypes.number.isRequired,
};

export default Footer;
