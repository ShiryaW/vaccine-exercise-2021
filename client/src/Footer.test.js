import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Footer from "./Footer";
import React from "react";
import { shallow } from "enzyme";

describe("Footer", () => {
  beforeEach(() => {
    configure({ adapter: new Adapter() });
  });

  it("renders the footer", () => {
    const dropdown = shallow(<Footer totalItems={1234} />);
    expect(dropdown).toMatchSnapshot();
  });
});
