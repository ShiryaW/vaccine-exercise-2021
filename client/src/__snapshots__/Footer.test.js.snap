// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the footer 1`] = `
<div
  className="footer"
>
  <p>
    Total items: 
    1234
  </p>
</div>
`;
